doc/docs/configuration.md

```diff
@@ -0,0 +1,445 @@
+
+When there is a type definition with a polymorphic variant, coq-of-ocaml transforms it to the closest inductive:
+```ocaml
+module Context = struct
+  type t = [`Dir | `Key]
+end
+```
+generates:
+```coq
+Module Context.
+  Inductive t : Set :=
+  | Dir : t
+  | Key : t.
+End Context.
+```
+When a constructor appears, this option helps to tell Coq from which module it is. For example:
+```ocaml
+let x : Context.t = `Dir
+```
+would be transformed to:
+```coq
+Definition x : Context.t := Dir.
+```
+which is incorrect. By giving the relation `["Dir", "Context.Dir"]`, we can tell coq-of-ocaml to generate the correct constructor with the correct module prefix:
+```coq
+Definition x : Context.t := Context.Dir.
+```
+
+## variant_types
+#### Example
+```
+"variant_types": [
+  ["Dir", "Context.key_or_dir"],
+  ["Key", "Context.key_or_dir"]
+]
+```
+
+#### Value
+A list of couples of a polymorphic variant constructor and a type name to use when the constructor appears.
+
+#### Explanation
+When we name a polymorphic variant with a type synonym:
+```ocaml
+type t = [`Dir | `Key]
+```
+an expression could still have a type without mentioning `t`:
+```ocaml
+let x : [`Dir] = `Dir
+```
+Since the polymorphic variants do not have a direct equivalent in OCaml, we could instead use a standard algebraic type:
+```ocaml
+type t = Dir | Key
+
+let x : t = Dir
+```
+which translates to:
+```coq
+Inductive t : Set :=
+| Dir : t
+| Key : t.
+
+Definition x : t := Dir.
+```
+
+When removing polymorphic variants is not possible, coq-of-ocaml transforms the type definition to the closest inductive type:
+```coq
+Inductive t : Set :=
+| Dir : t
+| Key : t.
+```
+and with the setting `["Dir", "t"]` it also correctly transforms the type of `x`:
+```coq
+Definition x : t := Dir.
+```
+
+## without_guard_checking
+#### Example
+```
+"without_guard_checking": [
+  "src/proto_alpha/lib_protocol/apply.ml",
+  "src/proto_alpha/lib_protocol/misc.ml",
+  "src/proto_alpha/lib_protocol/raw_context.ml",
+  "src/proto_alpha/lib_protocol/script_interpreter.ml",
+  "src/proto_alpha/lib_protocol/storage_description.ml"
+]
+```
+
+#### Value
+A list of filenames on which to disable termination checks by Coq.
+
+#### Explanation
+This option turns off the flag `Guard Checking`:
+```coq
+Unset Guard Checking.
+```
+Thus it is possible to write fixpoints which are not syntactically terminating. To help the evaluation tactics to terminate in the proofs, we can combine this setting with the [coq_struct](http://localhost:3000/coq-of-ocaml/docs/attributes#coq_struct) attribute.
```

Comment:
Can you explain more the part of adding the [coq_struct] into the code? I don't understand what does it mean about the evaluation tactics to terminate in the proof? And is the function still not termination even using the coq_struct? Is there any suggestion to help to fix the untermination functions, etc.

Reply:
Maybe you can give an example for this case as you did for the positivity check.

Reply:
Just added an example.